src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginTest.groovy

```diff
@@ -2,36 +2,279 @@ package io.github.cosmicsilence.scalafix
+    }
+
+    def 'checkScalafixMain task configuration validation'() {
+        setup:
+        applyScalafixPlugin(project, false, 'ExplicitResultTypes,DisableSyntax',
+                project.file('.scalafix.conf'))
+
+        when:
+        project.evaluate()
+
+        then:
+        ScalafixTask task = project.tasks.getByName('checkScalafixMain')
+        task.dependsOn.isEmpty()
+        task.configFile.get().asFile.path == "${project.projectDir}/.scalafix.conf"
+        task.rules.get().contains('ExplicitResultTypes')
+        task.rules.get().contains('DisableSyntax')
+    }
+
+    def 'checkScalafixMain task configuration validation when autoConfigureSemanticDb is enabled'() {
+        setup:
+        applyScalafixPlugin(project, true, 'ExplicitResultTypes,DisableSyntax',
+                project.file('.scalafix.conf'))
+
+        when:
+        project.evaluate()
+
+        then:
+        ScalafixTask task = project.tasks.getByName('checkScalafixMain')
+        task.dependsOn.find{ it.name == 'compileScala' }
+        task.configFile.get().asFile.path == "${project.projectDir}/.scalafix.conf"
+        task.rules.get().contains('ExplicitResultTypes')
+        task.rules.get().contains('DisableSyntax')
+    }
+
+    def 'checkScalafixTest task configuration validation'() {
+        setup:
+        applyScalafixPlugin(project, false, 'ExplicitResultTypes,DisableSyntax',
+                project.file('.scalafix.conf'))
+
+        when:
+        project.evaluate()
+
+        then:
+        ScalafixTask task = project.tasks.getByName('checkScalafixTest')
+        task.dependsOn.isEmpty()
+        task.configFile.get().asFile.path == "${project.projectDir}/.scalafix.conf"
+        task.rules.get().contains('ExplicitResultTypes')
+        task.rules.get().contains('DisableSyntax')
+    }
+
+    def 'checkScalafixTest task configuration validation when autoConfigureSemanticDb is enabled'() {
+        setup:
+        applyScalafixPlugin(project, true, 'ExplicitResultTypes,DisableSyntax',
+                project.file('.scalafix.conf'))
+
+        when:
+        project.evaluate()
+
+        then:
+        ScalafixTask task = project.tasks.getByName('checkScalafixTest')
+        task.dependsOn.find{ it.name == 'compileTestScala' }
+        task.configFile.get().asFile.path == "${project.projectDir}/.scalafix.conf"
+        task.rules.get().contains('ExplicitResultTypes')
+        task.rules.get().contains('DisableSyntax')
+    }
+
+    def 'scalafix task configuration validation'() {
+        when:
+        applyScalafixPlugin(project, false, 'ExplicitResultTypes,DisableSyntax',
+                project.file('.scalafix.conf'))
+
+        then:
+        Task task = project.tasks.getByName('scalafix')
+        task.dependsOn.find { it.name == 'scalafixMain' }
+        task.dependsOn.find { it.name == 'scalafixTest' }
+        !project.tasks.getByName('check').dependsOn.find { it.name == 'checkScalaFix' }
```

Comment:
Good catch! This test is actually working because that line is looking for `checkScalaFix` instead of `checkScalafix`. The right task would be `scalafix`.